kvdb-web/src/lib.rs

```diff
@@ -0,0 +1,169 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+//! A key-value database for use in browsers
+//!
+//! Writes data both into memory and IndexedDB, reads from the IndexedDB
+//! on `open`.
+
+
+mod error;
+mod indexed_db;
+
+use std::io;
+use std::rc::Rc;
+use std::sync::Mutex;
+use kvdb::{DBValue, DBTransaction};
+use kvdb_memorydb::{InMemory, self as in_memory};
+use send_wrapper::SendWrapper;
+
+pub use error::Error;
+pub use kvdb::KeyValueDB;
+
+use futures::prelude::*;
+
+use web_sys::IdbDatabase;
+
+pub struct Database {
```

Comment:
missing docs

Reply:
"Database" is a very generic name, although I guess `kvdb_web::Database` is readable. Other ideas: `WebDB`, `WebDatabase`, `IndexDB`, `WebKVDB`.

Reply:
I named it `Database` for consistency with `kvdb-memory` and `kvdb-rocksdb`, don't have a strong opinion on naming.